### PR TITLE
Skip KleidiCV in offline build

### PR DESCRIPTION
**Pull request by asmorkalov**
The PR reworks KleidiCV downloading and detection steps to make it similar to IPP and compatible with off-line build.
 
Fixes https://github.com/opencv/opencv/issues/26275

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [x] There is a reference to the original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -931,9 +931,12 @@ if(HAVE_OPENVX)
 endif()
 
 if(WITH_KLEIDICV)
-  ocv_debug_message(STATUS "Enable KleidiCV acceleration")
+  ocv_debug_message(STATUS "Try to enable KleidiCV acceleration")
   if(NOT ";${OpenCV_HAL};" MATCHES ";kleidicv;")
-    set(OpenCV_HAL "kleidicv;${OpenCV_HAL}")
+    add_subdirectory(3rdparty/kleidicv)
```
Comment (opencv-alalek) [inline CMakeLists.txt:936]: This skips `add_subdirectory()` call if user specifies `cmake -DOpenCV_HAL=kleidicv ...`.

**Comment by asmorkalov** [comment]
@opencv-alalek I patched conditions a bit.

File: CMakeLists.txt (hunk 1 above)
Comment (opencv-alalek) [reply CMakeLists.txt]: What is the problem to follow carotene design?

**Comment by opencv-alalek** [comment]
Carotene is here, but it requires NEON which could be unavailable.

**Comment by pavloshargan** [comment]
> In case, if NEON is not available Carotene is build as dummy library with "NOT_IMPLEMENTED" stubs. We cannot do the same if KleidiCV code is not available.
When is it not available?

I'm trying to build android sd on Mac m3, and getting Carotene-related errors like

```
 ld: error: undefined symbol: carotene_o4t::add(carotene_o4t::Size2D const&, unsigned char const*, long, unsigned char const*, long, unsigned char*, long, carotene_o4t::CONVERT_POLICY) ld: error: undefined symbol: carotene_o4t::isSupportedConfiguration()
 and so on
```
My build arguments are:
python3 build_sdk.py \
    ~/Downloads/opencv_sdk \
   ~/Downloads/opencv \
    --sdk_path ~/Library/Android/sdk \
    --ndk_path ~/Library/Android/sdk/ndk/25.1.8937393 \
    --config [ndk-25.config.py](http://ndk-25.config.py/) \
    --use_android_buildtools
   
    I tried specifically setting cmake_vars['WITH_CAROTENE'] = "ON" but it didn't change anything
   
    I see following outut during the build process:
    -- Android Gradle Plugin version: 7.3.1
-- Kotlin Plugin version: 1.8.20
-- Gradle version: 7.5.1
-- Android Gradle Java version: 1_8
-- Carotene: NEON is not available, disabling carotene...


    I'm building 71d3237a093b60a27601c20e9ee6c3e52154e8b1 (4.10 release)
   
    What am i doing wrong? My goal is to build opencv from source in order to have access to VideoCapture with Mediacodec backend

**Review by opencv-alalek — APPROVED** [review]

**Pull request merged**